3_harmonize/src/harmonize_sdd.R

```diff
@@ -13,483 +9,997 @@ harmonize_sdd <- function(raw_sdd, p_codes,
+    "bottom reading not taken", "afraid to touch bottom", "forgot bottom reading", "didn't hit bottom", "did not hit bottom",
+    "vegetation blocks", "turbid", "probe not on bottom", "sonde.*not.*bottom", "hydrolab not on bottom",
+    "profile not taken to bottom", "unable to collect bottom", "bottom.*not.*collected", "bottom.*not.*recorded",
+    sep = "|")
+  
+  # Approximate harmonized_value field with values from depth columns
+  sdd_approx_added <- sdd_greaterthan_added %>%
+    # clean the unit columns for consistency in comparisons
+    mutate(
+      across(all_of(unit_cols), ~ {
+        . <- tolower(trimws(.))
+        . <- gsub("feet", "ft", ., ignore.case = TRUE)
+        . <- gsub("meters", "m", ., ignore.case = TRUE)
+      }),
+      # create a bottom tag to detect bottom language detected in relevant comment columns to be used for flagging
+      bottom_tag = case_when(
+        rowSums(across(all_of(comment_cols), ~grepl("bottom", .x, ignore.case = TRUE))) > 0 ~ 1,
+        TRUE ~ 0
+      ),
+      # create a negate bottom tag to detect negate language in relevant comment columns
+      negate_bottom_tag = case_when(
+        rowSums(across(all_of(comment_cols), ~grepl(negate_bottom_text, .x, ignore.case = TRUE))) > 0 ~ 1,
+        TRUE ~ 0
+      ),
+      # flag 0 = value and units not adjusted
+      # flag 1 = value filled using activity depth measure column
+      # flag 2 = value filled using activity bottom depth measure column
+      approx_flag = case_when(
+        # If the harmonized value is not NA, set flag to 0
+        !is.na(harmonized_value) ~ 0,
+        # If the harmonized value is NA and there is bottom language and no negate language, and
+        # ActivityBottomDepthHeightMeasure.MeasureValue is not NA or 0 set flag to 2
+        is.na(harmonized_value) &
+          bottom_tag == 1 &
+          negate_bottom_tag == 0 &
+          !is.na(ActivityBottomDepthHeightMeasure.MeasureValue) &
+          ActivityBottomDepthHeightMeasure.MeasureValue != 0 ~ 2,
+        # If the harmonized value is NA and there is no bottom language or there is negate language, and
+        # ActivityDepthHeightMeasure.MeasureValue is not NA or 0 set flag to 1
+        is.na(harmonized_value) &
+          (bottom_tag == 0 | negate_bottom_tag == 1) &
+          !is.na(ActivityDepthHeightMeasure.MeasureValue) &
+          ActivityDepthHeightMeasure.MeasureValue != 0~ 1,
+        # For all other cases, set flag to NA
+        .default = NA_integer_ # it is possible to get NA values here, but we will drop them in the next step
+      ),
+      harmonized_value = case_when(
+        approx_flag == 2 ~ as.numeric(ActivityBottomDepthHeightMeasure.MeasureValue),
+        approx_flag == 1 ~ as.numeric(ActivityDepthHeightMeasure.MeasureValue),
+        .default = harmonized_value
+      ),
+      harmonized_units = case_when(
+        approx_flag == 2 ~ ActivityBottomDepthHeightMeasure.MeasureValue,
```

Comment:
Not sure, but I think this needs to be changed?
```suggestion
        approx_flag == 2 ~ ActivityBottomDepthHeightMeasure.MeasureUnitCode,
```

Reply:
Great catch!

Reply:
oops, this needs to be committed too!